ngraph/frontend/onnx_editor/include/onnx_editor/editor.hpp

```diff
@@ -110,11 +111,57 @@ namespace ngraph
             /// \param out_file_path A path to the file where the modified model should be dumped.
             void serialize(const std::string& out_file_path) const;
 
+            /// \brief Returns the InputEdge based on a node (node name or output name)
+            ///        and an input (input name or input index).
+            ///
+            /// \note  The node name can be ambiguous (many ONNX nodes can have the same name).
+            ///        In such a case the algorthim tries to match the given node name
+            ///        with the input name (providing an input index is not enough).
+            ///        If a unique edge is found, it will be returned.
+            ///        If InputEdge cannot be determined based on parameter values an ngraph_error
+            ///        exception will be thrown.
+            ///
+            /// \param node A node helper structure created based on a node name
+            ///             or a node output name.
+            ///
+            /// \param input An input helper structure created based on a input name
+            ///              or a input index.
+            InputEdge find_input_edge(const EditorNode& node, const EditorInput& input);
+
+            /// \brief Returns an OutputEdge based on a node (node name or output name)
+            ///        and an output (output name or output index).
+            ///
+            /// \note  The node name can be ambiguous (many ONNX nodes can have the same name).
+            ///        In such a case the algorthim will try to match the given node name
+            ///        with the output name (providing an output index is not enough).
+            ///        If after such operation a found edge is unique, it is returned.
+            ///        If OutputEdge cannot be determined based on given params the ngraph_error
+            ///        exception is thrown.
+            ///
+            /// \param node A node helper structure created based on a node name
+            ///             or a node output name.
+            ///
+            /// \param output A output helper structure created based on a output name
+            ///               or a output index.
+            OutputEdge find_output_edge(const EditorNode& node, const EditorOutput& output);
+
+            /// \brief Returns an OutputEdge based on a output name.
+            ///
+            /// \note  The output name guarantees the uniqueness of the edge.
+            ///
+            /// \param output_name A node output name.
+            ///
+            OutputEdge find_output_edge(const std::string& output_name);
```

Comment:
Another concern: I still don't see how to specify the input for subgraph extraction that is represented as an output of some node (a tensor that is produced). In MO it is specified as `tensor_name:0` given as `--input`.

Reply:
It seems to be: `find_input_edge(EditorNode{EditorOutput{"tensor_name"}}, EditorInput{0})` case or I miss sth?

Reply:
As we've discussed offline, `find_output_consumers` was introduced to handle such a case - https://github.com/openvinotoolkit/openvino/pull/4927/commits/53dabc6b603fd4f78edec31806e53de38814a29e